avocado/plugins/multiplex.py

```diff
@@ -35,12 +35,10 @@ def __init__(self, *args, **kwargs):
         super(Multiplex, self).__init__(*args, **kwargs)
 
     def configure(self, parser):
-        if multiplexer.MULTIPLEX_CAPABLE is False:
-            return
-
         parser = super(Multiplex, self).configure(parser)
-        parser.add_argument('multiplex', nargs='+',
-                            help='Path(s) to a multiplex file(s)')
+        if multiplexer.MULTIPLEX_CAPABLE:
+            parser.add_argument("-m", '--multiplex', nargs='*',
```

Comment:
OK, I see this change as a way to work around a design decision that is not optimal.  Let me try to explain.

1) The more natural way to *use* the multiplexer command is to run `avocado multiplex foo.yaml` or `avocado multiplex foo.json` supposing JSON is a valid "multiplex plugin".

2) The reason for turning this from a positional argument to a non positional is such it can be replaced by the proposed plugins, and forced into the command line.  Example of your current implementation changes this argument to this one:

```
yaml to mux options:
  -m [FILE [FILE ...]], --multiplex [FILE [FILE ...]]
                        Location of one or more Avocado multiplex (.yaml)
                        FILE(s) (order dependent)
```

When that `CLI` plugin is `configure()d`.  That is, the plugin code is setting parameters so that it can tell the world that it exists, and while not the case here, get the parameters it needs and then act on them (turn the YAML into a MUX).

3) One proof that this is unoptimal, is that it wouldn't allow the user to have a secondary mux plugin installed.

4) The ideal design, would be one that the framework selects (somehow) which is the Mux plugin to be active, and then the *basic* command line options (such as the positional arguments to the Mux "source") wouldn't be changed.  Additional options related to a given Mux plugin implementation of couse could be added.

That is, the Mux interface would be similar to what we eventually have to do with the runner.  For a given job (or test) a given runner would be selected by the framework.  The basic parameters, say the test references, would remain positional arguments.

Of couse we lack this easy-to-use way of having "driver style" plugins, but IMHO, we'll need that to have the correct design implemented.

Does it make sense?